[#1625] Bugfix/mqtt yield

TD-er opened the pull request:
Also did some clean-up in the Scheduler.
Uses of these scheduled event handling:

- MQTT import (P037)
- MQTT callback => `CPLUGIN_PROTOCOL_RECV`
- Notifications

These issues might also be fixed by this:

- #1213 
- #1254

TD-er committed:
[Scheduler] Rename some functions and clean-up

TD-er committed:
[Notifications] Schedule NPlugin_ptr calls async

To make sure execution of these functions is called from the `loop()` and not from a callback function.

TD-er committed:
[MQTT] Schedule from callback functions

To make sure execution of these functions is called from the `loop()` and not from a callback function.

thomastech commented:
I'll test it tonight. The branch is still in-progress, so I'll come back later.
- Thomas

TD-er committed:
[Rules] Split parsing single line from rulesProcessingFile 

The function `rulesProcessingFile` is way too long to understand.
First step in making those functions readable.

TD-er committed:
[Rules] Split processing matched rule in a separate function

Just to get rules processing readable

TD-er committed:
[Commands] Return String output per command

The existing implementation was a bit sparse in returned info.
Now each command can return a useful string to display in the command output window in the webinterface. (or whatever the source of the command is)

TD-er committed:
[Schedule] Scheduled command timer

Run commands async. The result is now not displayed in the web page anymore.

TD-er committed:
[Schedule] Run command from webpage immediately

When sending command request from the Tools tab in the web interface, the command will be executed immediately.
In other situations it will use the scheduler to run commands async.

Commands will now be buffered and processed in order.
This will also speed-up rules which do a publish.
